fe/fe-core/src/main/java/com/starrocks/statistic/AnalyzeManager.java

```diff
@@ -147,4 +155,77 @@ public void write(DataOutput out) throws IOException {
         @SerializedName("analyzeJobs")
         public List<AnalyzeJob> jobs;
     }
+
+    // This task is used to expire cached statistics
+    public class AnalyzeReplayTask implements Runnable {
+        private AnalyzeJob analyzeJob;
+
+        public AnalyzeReplayTask(AnalyzeJob job) {
+            this.analyzeJob = job;
+        }
+
+        public void checkAndExpireCachedStatistics(Table table, AnalyzeJob job) {
+            if (null == table || !Table.TableType.OLAP.equals(table.getType())) {
+                return;
+            }
+
+            // check table has update
+            // use job last work time compare table update time to determine whether to expire cached statistics
+            LocalDateTime updateTime = StatisticUtils.getTableLastUpdateTime(table);
+            LocalDateTime jobLastWorkTime = LocalDateTime.MIN;
+            if (analyzeJobMap.containsKey(job.getId())) {
+                jobLastWorkTime = analyzeJobMap.get(job.getId()).getWorkTime();
+            }
+            if (jobLastWorkTime.isBefore(updateTime)) {
+                List<String> columns = (job.getColumns() == null || job.getColumns().isEmpty()) ?
+                        table.getFullSchema().stream().filter(d -> !d.isAggregated()).map(Column::getName)
+                                .collect(Collectors.toList()) : job.getColumns();
+                Catalog.getCurrentStatisticStorage().expireColumnStatistics(table, columns);
+            }
+        }
+
+        public void expireCachedStatistics(AnalyzeJob job) {
+            if (job.getScheduleType().equals(Constants.ScheduleType.ONCE)) {
+                Database db = Catalog.getCurrentCatalog().getDb(job.getDbId());
+                if (null == db) {
+                    return;
+                }
+                Catalog.getCurrentStatisticStorage()
+                        .expireColumnStatistics(db.getTable(job.getTableId()), job.getColumns());
+            } else {
+                if (job.getDbId() == AnalyzeJob.DEFAULT_ALL_ID) {
```

Comment:
These if else clauses  could unify,  select all tables firstly, then `checkAndExpireCachedStatistics` for one table